### PR TITLE
Add Health Endpoint

### PR DESCRIPTION
**Pull request by usingtechnology**
Add basic health check (check logstash only).
Add unit tests.
Update CLOGS morgan parser to handle array (bug fix, was not updated when we moved from single message log to array of messages).

# Description

[SHOWCASE-649](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-649)
## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the OpenAPI 3.0 `v*.api-spec.yaml` documentation (if appropriate)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by usingtechnology** [comment]
This is up at: https://clogs-pr-8.pathfinder.gov.bc.ca/

I've massaged the old field counts (they were assuming a single logging message, not a batch). So that's included in this release.  The following is the CLOGS clogs structure.

`    "clogs": {
      "env": "dev",
      "client": "CLOGS",
      "data": {
        "azp": "CDOGS_SERVICE_CLIENT",
        "items": {
          "fields": {
            "message": "2",
            "data": "1",
            "pattern": "1",
            "level": "1"
          },
          "count": "3"
        },
        "op": "LOG",
        "responseTime": "157.233",
        "ts": "1582218292225",
        "httpStatus": "201"
      },
      "level": "info",
      "timestamp": 1582218292382,
      "retention": "default",
      "@timestamp": "2020-02-20T17:04:52.382Z"
    },
`

File: app/src/components/healthSvc.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+const utils = require('./utils');
+
+const healthSvc = {
+  /**
+     *  @function check
+     *  Call the health checks.
+     *  1. Call logstash node stats events API to see if we can connect to logstash
+     *  Return health check object array
+     */
```
Comment (jujaga) [inline app/src/components/healthSvc.js]: Two extra spaces overindented

File: app/src/middleware/apiTracker.js [hunk 2]
```diff
@@ -25,13 +25,33 @@ const trackerUrls = [logUrl];
...
+  // move all the counts into a sub-object of data...
+  clogsMessage.clogs.data.items = {
+    count: parsed.clogs.data.logItemsCount,
+    fields: {
+      message: parsed.clogs.data.messageFieldCount,
+      data: parsed.clogs.data.dataFieldCount,
+      pattern: parsed.clogs.data.patternFieldCount,
+      level: parsed.clogs.data.levelFieldCount
+    }
+  };
```
Comment (jujaga) [inline app/src/middleware/apiTracker.js]: With how many times we're pulling from parsed.clogs.data, may be better to have a const data = parsed.clogs.data assignment after line 35.

File: app/src/middleware/apiTracker.js [hunk 3]
```diff
@@ -87,20 +113,24 @@ const initializeApiTracker = app => {
     return req._ts ? req._ts : '0';
   });
 
+  morgan.token('count', req => {
+    return req.body && Array.isArray(req.body) ? req.body.length : '0';
```
Comment (jujaga) [inline app/src/middleware/apiTracker.js:118]: Is that 0 meant to be a string or an integer?

File: openshift/README.md [hunk 4]
```diff
@@ -122,5 +122,5 @@ oc -n <namespace> rollout latest dc/<buildname>-master
 As of this time, we do not automatically clean up resources generated by a Pull Request once it has been accepted and merged in. This is still a manual process. Our PR deployments are all named in the format "pr-###", where the ### is the number of the specific PR. In order to clear all resources for a specific PR, run the following two commands to delete all relevant resources from the Openshift project (replacing `PRNUMBER` with the appropriate number):
 
 ```sh
-oc delete all -n <namespace> --selector app=clogs-pr-<PRNUMBER>
+oc delete all -n <namespace> --selector app=common-logging-<job name>
```
Comment (jujaga) [inline openshift/README.md:125]: Is this correct? PR-8 still has its pods with the label `app: clogs-pr-8`.

**Review by jujaga — CHANGES_REQUESTED** [review]

**Review by loneil — APPROVED** [review]
Looks good

File: app/src/middleware/apiTracker.js (hunk 2 above)
Comment (usingtechnology) [reply app/src/middleware/apiTracker.js]: 👍

File: openshift/README.md (hunk 4 above)
Comment (usingtechnology) [reply openshift/README.md:125]: ha, that was me just using the incorrect APP_NAME on the command line...

File: app/src/middleware/apiTracker.js (hunk 3 above)
Comment (usingtechnology) [reply app/src/middleware/apiTracker.js:118]: It has to be a string, if it is a number, then morgan treats it as a boolean and places '-' as the value.

**Review by jujaga — APPROVED** [review]

File: app/src/components/healthSvc.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+   *  Return health check object array
+   */
+  check: async () => {
+    const logstash = {
+      name: 'logstash',
+      healthy: false,
+      info: ''
+    };
+    try {
+      const response = await axios.get(config.get('elkStack.logstashUrl'));
```
Comment (TimCsaky) [inline app/src/components/healthSvc.js:21]: bit late here and i'm probably missing something but are you calling the logstashUrl ? isnt the node stats events API at /_node/stats/events?pretty'

**Review by TimCsaky — APPROVED** [review]
looks good

File: app/src/components/healthSvc.js (hunk 5 above)
Comment (usingtechnology) [reply app/src/components/healthSvc.js:21]: yeah, good catch.  will have to change that comment next PR.
i was going to hit stats, but the configured logstash url is to hit the http port, and we dont have the exposed _node/stats.  So hitting the /http is good enough for now to see that we can connect.

Comment (TimCsaky) [reply app/src/components/healthSvc.js:21]: yeah i dont see the difference

**Pull request merged**